plugins/auth-backend/src/providers/gitlab/provider.ts

```diff
@@ -47,29 +57,58 @@ type PrivateInfo = {
 
 export type GitlabAuthProviderOptions = OAuthProviderOptions & {
   baseUrl: string;
+  signInResolver?: SignInResolver<OAuthResult>;
+  authHandler?: AuthHandler<OAuthResult>;
+  tokenIssuer: TokenIssuer;
+  catalogIdentityClient: CatalogIdentityClient;
+  logger: Logger;
 };
 
-function transformProfile(fullProfile: FullProfile) {
-  const profile = makeProfileInfo({
-    ...fullProfile,
-    photos: [
-      ...(fullProfile.photos ?? []),
-      ...(fullProfile.avatarUrl ? [{ value: fullProfile.avatarUrl }] : []),
-    ],
-  });
+export const extractGitLabUserId = (profile: ProfileInfo): string => {
+  return profile.username || (profile.email?.split('@')[0] as string);
+};
+
+export const gitlabDefaultSignInResolver: SignInResolver<OAuthResult> = async (
```

Comment:
The default sign-in resolver should mimic the existing implementation in order for this not to be a breaking change, so it should split the email if available and otherwise use the `info.result.fullProfile.id`. Yeah it's a bit backwards but it will go away 😅

Reply:
I added this as we use the username to match to the users in backstage, the e-mail is often different (or formatted differently between users) and can be changed easily.

I can change this to use the id to avoid a breaking change but we would need the username for our login handler.
Can we add additional profile data on the profile here (e.g. as a `data` prop) or would that be considered a breaking change as well?